src/Kestrel.Core/Internal/Http2/HPack/HPackDecoder.cs

```diff
@@ -108,11 +109,15 @@ internal HPackDecoder(int maxDynamicTableSize, DynamicTable dynamicTable)
             _dynamicTable = dynamicTable;
         }
 
-        public void Decode(Span<byte> data, bool endHeaders, IHttpHeadersHandler handler)
+        public void Decode(ReadOnlySequence<byte> data, bool endHeaders, IHttpHeadersHandler handler)
         {
-            for (var i = 0; i < data.Length; i++)
+            foreach (var segment in data)
```

Comment:
This is going to be inefficient, but I'm not sure it's worth it to fix this until we remove OnByte